core/trino-spi/src/main/java/io/trino/spi/connector/BasicRelationStatistics.java

```diff
@@ -0,0 +1,43 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.trino.spi.connector;
+
+public class BasicRelationStatistics
+{
+    /**
+     * Estimated row count
+     */
+    private final long rowCount;
+
+    /**
+     * Estimated data size in bytes
+     */
+    private final long dataSize;
```

Comment:
Why not use `double` for the estimates?
Would make it consistent with other stats-related APIs.

Just document on getter and validate in ctor than NaN values are forbidden

Reply:
I can go with double. I was on the fence here.